user.php

```diff
@@ -0,0 +1,108 @@
+<?
+
+
+class User{
+
+	public $db;
+
+	private $states = [
+		'NSW',
+		'ACT',
+		'VIC',
+		'QLD',
+		'TAS',
+		'WA'
+	]
+
+	function __construct()
+	{
+		$this->db = new mysqli('127.0.0.1', 'root', 'password', 'users');
+
+	}
+
+	function check_Postcode($postcode) {
+		//.... call some external api
+		$res = external_api_call();
+		if (!$res) {
+			throw new \Exception("invalid post code");
+		}
+		return true;
+	}
+
+
+	function register($name, $password, $email, $phone = null, $postal_address = null, $postalState = null
+			, $billing_address = null, $billingState = null)
+	{
+		global $validator;
```

Comment:
The use of global variables must be avoided. This validator should be injected as a dependency in this class.